pkg/scheduler/framework/runtime/framework.go

```diff
@@ -613,7 +613,10 @@ func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framewor
 	}()
 	var result *framework.PreFilterResult
 	var pluginsWithNodes []string
+	logger := klog.FromContext(ctx).WithName("PreFilter").WithValues("pod", klog.KObj(pod))
```

Comment:
Would be great to validate all the pod, nodes and other keys in the klog.Infos that got removed are properly replaced by `WithName` and `WithValues`. E.g. checking the logs and comparing the lines before and after.

Reply:
Right now, log lines should still include all extra values that are important, i.e. nothing should have been removed. That is because contextual logging is an optional feature and currently (because still alpha) disabled by default:

https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/3077-contextual-logging/README.md#redundant-keyvalue-pairs

Later when this is GA, the code can be simplified by removing redundant key/value pairs that are guaranteed to set via WithValues.

I agree that comparing logs would be useful, for both cases (contextual logging enabled and disabled). The problem will be that a simple diff will not work.

Reply:
I will compile and send here a comparison between 3 options:

- what we had before this PR
- What we have with this PR, but contextual logging enabled
- What we have with this PR, but contextual logging disabled

Would that be sufficient?

Reply:
SIG Scheduling is the first SIG to deal with this. The recommendation from the KEP is to keep the potentially duplicated key/value pairs. They will be filtered out by klog at runtime.